presto-main/src/main/java/com/facebook/presto/sql/planner/sanity/NoExpressionLeftCheck.java

```diff
@@ -0,0 +1,151 @@
+    }
+
+    private static class Visitor
+            extends SimplePlanVisitor<Void>
+    {
+        @Override
+        public Void visitProject(ProjectNode node, Void context)
+        {
+            visitPlan(node, context);
+
+            for (RowExpression expression : node.getAssignments().getExpressions()) {
+                if (isExpression(expression)) {
+                    throw new IllegalStateException();
+                }
+            }
+            return null;
+        }
+
+        @Override
+        public Void visitAggregation(AggregationNode node, Void context)
+        {
+            visitPlan(node, context);
+
+            for (AggregationNode.Aggregation aggregation : node.getAggregations().values()) {
+                for (RowExpression expression : aggregation.getArguments()) {
+                    if (isExpression(expression)) {
+                        throw new IllegalStateException(ERROR_MESSAGE);
+                    }
+                }
+                if (aggregation.getFilter().isPresent() && isExpression(aggregation.getFilter().get())) {
+                    throw new IllegalStateException(ERROR_MESSAGE);
+                }
+            }
+            return null;
+        }
+
+        @Override
+        public Void visitWindow(WindowNode node, Void context)
+        {
+            visitPlan(node, context);
+
+            for (WindowNode.Function function : node.getWindowFunctions().values()) {
+                for (RowExpression expression : function.getFunctionCall().getArguments()) {
+                    if (isExpression(expression)) {
+                        throw new IllegalStateException(ERROR_MESSAGE);
+                    }
+                }
+            }
+            return null;
+        }
+
+        @Override
+        public Void visitFilter(FilterNode node, Void context)
+        {
+            visitPlan(node, context);
+
+            if (isExpression(node.getPredicate())) {
+                throw new IllegalStateException(ERROR_MESSAGE);
+            }
+            return null;
+        }
+
+        @Override
+        public Void visitValues(ValuesNode node, Void context)
+        {
+            visitPlan(node, context);
+
+            for (List<RowExpression> rows : node.getRows()) {
+                for (RowExpression row : rows) {
+                    if (isExpression(row)) {
+                        throw new IllegalStateException(ERROR_MESSAGE);
+                    }
+                }
+            }
+            return null;
+        }
+
+        @Override
+        public Void visitJoin(JoinNode node, Void context)
+        {
+            visitPlan(node, context);
+
+            if (node.getFilter().isPresent() && isExpression(node.getFilter().get())) {
+                throw new IllegalStateException(ERROR_MESSAGE);
+            }
+            return null;
+        }
+
+        @Override
+        public Void visitSpatialJoin(SpatialJoinNode node, Void context)
```

Comment:
`Apply`, `TableFinish` and `TableWriter` is missing. Check `ExpressionTranslations` optimization rule for the list of nodes.